litgpt/finetune/adapter_v2.py

```diff
@@ -146,14 +150,23 @@ def main(
     model = fabric.setup_module(model)
 
     if isinstance(fabric.strategy.precision, BitsandbytesPrecision):
-        import bitsandbytes as bnb
+        if (isinstance(optimizer, str) and "AdamW" not in optimizer) or (isinstance(optimizer, dict) and "AdamW" not in optimizer.get("class_path", "")):
+            raise ValueError("The chosen quantization format only supports the AdamW optimizer.")
 
-        optimizer_cls = bnb.optim.PagedAdamW
+        import bitsandbytes as bnb
+        if isinstance(optimizer, str):
+            optimizer = bnb.optim.PagedAdamW(model.parameters())
+        else:
+            optim_args = get_argument_names(bnb.optim.PagedAdamW)
+            allowed_kwargs = {key: optimizer["init_args"][key] for key in optim_args & optimizer["init_args"].keys()}
+            optimizer = bnb.optim.PagedAdamW(model.parameters(), **allowed_kwargs)
```

Comment:
I would create a function out of the bitsandbytes optimizer creation code path and another one for else content (L164-168) since this is going to be the same code everywhere